### PR TITLE
add AC02 cards

### PR DESCRIPTION
**Pull request by Trishula9**

### CODE REVIEW

File: scripts/AC02-JP/c100290006.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	if chk==0 then return Duel.CheckReleaseGroup(tp,c100290006.rfilter,1,nil,tp) end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_RELEASE)
+	local g=Duel.SelectReleaseGroup(tp,c100290006.rfilter,1,1,nil,tp)
+	Duel.Release(g,REASON_COST)
+end
+function c100290006.spfilter(c,e,tp)
+	return c.toss_dice and c:IsCanBeSpecialSummoned(e,0,tp,false,false)
+end
+function c100290006.target(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100290006.spfilter,tp,LOCATION_HAND+LOCATION_DECK,0,1,nil,e,tp) end
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290006.lua:35]:
```suggestion
	if chk==0 then
		if e:GetLabel()~=1 then return false end
		e:SetLabel(0)
		return Duel.IsExistingMatchingCard(c100290006.spfilter,tp,LOCATION_HAND+LOCATION_DECK,0,1,nil,e,tp)
	end
```

File: scripts/AC02-JP/c100290006.lua [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	return c.toss_dice and c:IsFaceup()
+end
+function c100290006.condition(e,tp,eg,ep,ev,re,r,rp)
+	return Duel.IsExistingMatchingCard(c100290006.filter,tp,LOCATION_ONFIELD,0,1,nil)
+end
+function c100290006.rfilter(c,tp)
+	return Duel.GetMZoneCount(tp,c)>0
+end
+function c100290006.cost(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.CheckReleaseGroup(tp,c100290006.rfilter,1,nil,tp) end
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290006.lua:26]:
```suggestion
	e:SetLabel(1)
	if chk==0 then return Duel.CheckReleaseGroup(tp,c100290006.rfilter,1,nil,tp) end
```

File: scripts/AC02-JP/c100290006.lua [hunk 3]
```diff
@@ -0,0 +1,45 @@
+--ディメンジョン・ダイス 
+--
+--Script by Trishula9
+function c100290006.initial_effect(c)
+	--activate
+	local e1=Effect.CreateEffect(c)
+	e1:SetCategory(CATEGORY_SPECIAL_SUMMON)
+	e1:SetType(EFFECT_TYPE_ACTIVATE)
+	e1:SetCode(EVENT_FREE_CHAIN)
+	e1:SetCondition(c100290006.condition)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290006.lua:10]:
```suggestion
	e1:SetLabel(0)
	e1:SetCondition(c100290006.condition)
```

**Review by Lyris12 — CHANGES_REQUESTED** [review]
While you have no available Monster Zones, no effect should be able to copy the effect of "Dimension Dice" when the latter name activates.

File: scripts/AC02-JP/c100290010.lua [hunk 4]
```diff
@@ -0,0 +1,75 @@
+--A宝玉獣 ルビー・カーバンクル
+--
+--Script by Trishula9
+function c100290010.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290010.lua:4]:
```suggestion
function c100290010.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290011.lua [hunk 5]
```diff
@@ -0,0 +1,63 @@
+--A宝玉獣 アメジスト・キャット
+--
+--Script by Trishula9
+function c100290011.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290011.lua:4]:
```suggestion
function c100290011.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290012.lua [hunk 6]
```diff
@@ -0,0 +1,66 @@
+--A宝玉獣 エメラルド・タートル
+--
+--Script by Trishula9
+function c100290012.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290012.lua:4]:
```suggestion
function c100290012.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290013.lua [hunk 7]
```diff
@@ -0,0 +1,66 @@
+--A宝玉獣 トパーズ・タイガー
+--
+--Script by Trishula9
+function c100290013.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290013.lua:4]:
```suggestion
function c100290013.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290014.lua [hunk 8]
```diff
@@ -0,0 +1,84 @@
+--A宝玉獣 アンバー・マンモス
+--
+--Script by Trishula9
+function c100290014.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290014.lua:4]:
```suggestion
function c100290014.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290015.lua [hunk 9]
```diff
@@ -0,0 +1,100 @@
+--A宝玉獣 コバルト・イーグル
+--
+--Script by Trishula9
+function c100290015.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290015.lua:4]:
```suggestion
function c100290015.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

File: scripts/AC02-JP/c100290016.lua [hunk 10]
```diff
@@ -0,0 +1,76 @@
+--A宝玉獣 サファイア・ペガサス
+--
+--Script by Trishula9
+function c100290016.initial_effect(c)
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290016.lua:4]:
```suggestion
function c100290016.initial_effect(c)
	Duel.EnableGlobalFlag(GLOBALFLAG_SELF_TOGRAVE)
```

**Review by Lyris12 — CHANGES_REQUESTED** [review]
The effect code, `EFFECT_SELF_TOGRAVE`, requires `GLOBALFLAG_SELF_TOGRAVE` to be enabled in that duel to work.

**Comment by Trishula9** [comment]
> The effect code, `EFFECT_SELF_TOGRAVE`, requires `GLOBALFLAG_SELF_TOGRAVE` to be enabled in that duel to work.

Thank you for bug resolution

File: scripts/AC02-JP/c100290014.lua [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_SINGLE)
+	e2:SetCode(EFFECT_TO_GRAVE_REDIRECT_CB)
+	e2:SetProperty(EFFECT_FLAG_UNCOPYABLE)
+	e2:SetCondition(c100290014.repcon)
+	e2:SetOperation(c100290014.repop)
+	c:RegisterEffect(e2)
+	--negate
+	local e3=Effect.CreateEffect(c)
+	e4:SetDescription(aux.Stringid(100290014,0))
```
Comment (Lyris12) [inline scripts/AC02-JP/c100290014.lua:24]:
```suggestion
	e3:SetDescription(aux.Stringid(100290014,0))
```

**Comment by chronogenexx** [comment]
Doodle Beast and Doodlebook's japanese name is mistaken.

and, it would be better to use this:
```
#!setname 0x286 Doodle	らくがき
!setname 0x1286 Doodle Beast	らくがきじゅう
!setname 0x2286 Doodlebook	らくがきちょう
```

**Pull request merged**